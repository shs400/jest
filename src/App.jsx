import React, { Component } from 'react';
import Counter from './components/Counter';
import NameForm from './components/NameForm';
import NameList from './components/NameList';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      names: ['로이', '심현수']
    }
  }

  onInsert = (name) => {
    this.setState(({names}) => ({ names: names.concat(name) }));
  };

  render() {
    const { names } = this.state;
    const { onInsert } = this;

    return (
      <div>
        <Counter />
        <hr />
        <h1>이름 목록</h1>
        <NameForm onInsert={onInsert}/>
        <NameList names={names}/>
      </div>
    );
  }
}

export default App;
