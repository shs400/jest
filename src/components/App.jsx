import React, { Component } from 'react';
import Counter from './Counter/index';
import NameForm from './NameForm/index';
import NameList from './NameList/index';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      names: ['로이', '심현수'],
    };
  }

  onInsert = (name) => {
    this.setState(({ names }) => ({ names: names.concat(name) }));
  };

  render() {
    const { names } = this.state;

    return (
      <div>
        <Counter />
        <hr />
        <h1>이름 목록</h1>
        <NameForm onInsert={this.onInsert} />
        <NameList names={names} />
      </div>
    );
  }
}

export default App;
