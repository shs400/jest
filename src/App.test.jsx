import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('App', () => {
  let component = null;

  it('renders correctly', () => {
    component = renderer.create(<App />);
  });

  it('matches snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/*
  describe 와 it 에서 첫번째 파라미터는 작업의 설명을 넣어주게 되는데,
  describe 에서는 어떤 기능을 확인하는지, 그리고 it 부분에선 무엇을 검사해야 되는지에 대한 설명을 넣으시면 됩니다.
*/