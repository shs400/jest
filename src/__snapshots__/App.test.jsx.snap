// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div>
  <div>
    <h1>
      카운터
    </h1>
    <h2>
      1
    </h2>
    <button
      onClick={[Function]}
    >
      +
    </button>
    <button
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <hr />
  <h1>
    이름 목록
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label>
      이름
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      등록
    </button>
  </form>
  <ul>
    <li>
      로이
    </li>
    <li>
      심현수
    </li>
  </ul>
</div>
`;
